android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -383,28 +430,63 @@ private void initializePlayer() {
     }
 
     private MediaSource buildMediaSource(Uri uri, String overrideExtension) {
-        int type = Util.inferContentType(!TextUtils.isEmpty(overrideExtension) ? "." + overrideExtension
-                : uri.getLastPathSegment());
+        DataSource.Factory ds;
+        String userAgent = null;
+        if(userAgent == null || userAgent.isEmpty())
+            userAgent = DataSourceUtil.getUserAgent(themedReactContext);
+
+            try {
+                final RawResourceDataSource raw = new RawResourceDataSource(this.themedReactContext);
+                raw.open(new DataSpec(uri));
+                ds = new DataSource.Factory() {
+                    @Override
+                    public DataSource createDataSource() {
+                        return raw;
+                    }
+                };
+            } catch(IOException ex) {
+                // Should never happen
+                // throw new RuntimeException(ex);
+            }
+            // Creates a default http source factory, enabling cross protocol redirects
+            ds = new DefaultHttpDataSourceFactory(
+                    userAgent, null,
+                    DefaultHttpDataSource.DEFAULT_CONNECT_TIMEOUT_MILLIS,
+                    DefaultHttpDataSource.DEFAULT_READ_TIMEOUT_MILLIS,
+                    true
+            );
+
+        ds = enableCaching(ds);
+
+
+        String type = "default";
+
+        String[] formats = new String[]{"dash", "hls", "smoothstreaming"};
+
+        // Convert String Array to List
+        List<String> list = Arrays.asList(formats);
+        System.out.println(overrideExtension);
+        if(list.contains(overrideExtension)){
+            type = overrideExtension;
+        }
+
         switch (type) {
-            case C.TYPE_SS:
-                return new SsMediaSource(uri, buildDataSourceFactory(false),
-                        new DefaultSsChunkSource.Factory(mediaDataSourceFactory), 
-                        minLoadRetryCount, SsMediaSource.DEFAULT_LIVE_PRESENTATION_DELAY_MS, 
-                        mainHandler, null);
-            case C.TYPE_DASH:
-                return new DashMediaSource(uri, buildDataSourceFactory(false),
-                        new DefaultDashChunkSource.Factory(mediaDataSourceFactory), 
-                        minLoadRetryCount, DashMediaSource.DEFAULT_LIVE_PRESENTATION_DELAY_MS,
-                        mainHandler, null);
-            case C.TYPE_HLS:
-                return new HlsMediaSource(uri, mediaDataSourceFactory, 
-                        minLoadRetryCount, mainHandler, null);
-            case C.TYPE_OTHER:
-                return new ExtractorMediaSource(uri, mediaDataSourceFactory, new DefaultExtractorsFactory(),
+            case "smoothstreaming":
```

Comment:
We should not be scrapping the existing media source creators. They contain lots of configuration options that people need.